src/components/structures/RoomStatusBar.tsx

```diff
@@ -262,7 +342,98 @@ export default class RoomStatusBar extends React.PureComponent<IProps, IState> {
         />;
     }
 
-    public render(): JSX.Element {
+    private getRefreshTimelineContent(): JSX.Element {
+        let buttonRow = <>
+            <AccessibleButton
+                onClick={this.onRefreshTimelineClick}
+                className="mx_RoomStatusBar_refreshTimelineBtn"
+                data-testid="refresh-timeline-button"
+            >
+                { _t("Refresh timeline") }
+            </AccessibleButton>
+        </>;
+        if (this.state.isRefreshing) {
+            buttonRow = <>
+                <InlineSpinner w={20} h={20} />
+                { /* span for css */ }
+                <span>{ _t("Refreshing") }</span>
+            </>;
+        }
+
+        let errorContent: JSX.Element = <></>;
+        if (this.state.refreshError) {
+            let errorTextContent;
+            let submitDebugLogsTextContent;
+            if (this.state.refreshError.name === "ConnectionError") {
```

Comment:
See the pattern we ended up with in https://github.com/matrix-org/matrix-react-sdk/pull/10405